From 3cb6a14614a85c1fa6f68e3cde1d19e84325e132 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Wed, 26 May 2010 15:00:24 +0000
Subject: [panel] Make sure that newly-created panels become unhidden

The patch from a7412dfa introduces a bug where new panels will never get
notified that they should start doing their initial animation; they'll
always stay hidden.
---
diff --git a/gnome-panel/applet.c b/gnome-panel/applet.c
index 7c36e87..ee16df5 100644
--- a/gnome-panel/applet.c
+++ b/gnome-panel/applet.c
@@ -1033,18 +1033,18 @@ panel_applet_compare (const PanelAppletToLoad *a,
 void
 panel_applet_load_queued_applets (gboolean initial_load)
 {
-	if (initial_load) {
-		if (!panel_applets_to_load) {
-			panel_applet_queue_initial_unhide_toplevels (NULL);
-			return;
-		} else {
-			/* Install a timeout to make sure we don't block the
-			 * unhiding because of an applet that doesn't load */
-			panel_applet_unhide_toplevels_timeout =
-				g_timeout_add_seconds (UNHIDE_TOPLEVELS_TIMEOUT_SECONDS,
-						       panel_applet_queue_initial_unhide_toplevels,
-						       NULL);
-		}
+	if (!panel_applets_to_load) {
+		panel_applet_queue_initial_unhide_toplevels (NULL);
+		return;
+	}
+
+	if (initial_load && panel_applets_to_load) {
+		/* Install a timeout to make sure we don't block the
+		 * unhiding because of an applet that doesn't load */
+		panel_applet_unhide_toplevels_timeout =
+			g_timeout_add_seconds (UNHIDE_TOPLEVELS_TIMEOUT_SECONDS,
+					       panel_applet_queue_initial_unhide_toplevels,
+					       NULL);
 	}
 
 	panel_applets_to_load = g_slist_sort (panel_applets_to_load,
--
cgit v0.8.3.1
